gluish/common.py

```diff
@@ -64,3 +130,215 @@ def run(self):
+    branch = luigi.Parameter(default="master", significant=False)
+
+    def requires(self):
+        return [
+            GitCloneRepository(gitrepository=self.gitrepository,
+                               repositorydirectory=self.repositorydirectory,
+                               basedirectory=self.basedirectory),
+            Executable(name='git')
+        ]
+
+    def run(self):
+        currentdir = os.curdir
+
+        os.chdir(self.output().path)
+        shellout("""git checkout {branch}""", branch=self.branch)
+        shellout("""git pull origin {branch}""", branch=self.branch)
+
+        os.chdir(currentdir)
+
+    def complete(self):
+        if not self.output().exists():
+            return False
+
+        currentdir = os.curdir
+
+        os.chdir(self.output().path)
+        output = shellout("""git fetch origin {branch} > {output} 2>&1""", branch=self.branch)
+
+        result = True
+
+        with open(output, 'rb') as fh:
+            content = fh.readlines()
+            if len(content) >= 3:
+                result = False
+
+        revparseoutput = shellout("""git rev-parse {branch} > {output} 2>&1""", branch=self.branch)
+        originrevparseoutput = shellout("""git rev-parse origin/{branch} > {output} 2>&1""", branch=self.branch)
+
+        revparse = getfirstline(revparseoutput, "0")
+        originrevparse = getfirstline(originrevparseoutput, "1")
+
+        if revparse != originrevparse:
+            result = False
+
+        os.chdir(currentdir)
+
+        return result
+
+    def output(self):
+        return luigi.LocalTarget(path=os.path.join(str(self.basedirectory), str(self.repositorydirectory)))
+
+
+class CleanSolrIndex(luigi.Task):
+    """
+    Cleans a given Solr index from already existing data of a data package
+    """
+    date = ClosestDateParameter(default=datetime.date.today())
+    solruri = luigi.Parameter()
+    solrcore = luigi.Parameter()
+    sourceid = luigi.Parameter()
+    taskdir = luigi.Parameter()
+    salt = luigi.Parameter()
+    uritemplate = '{solruri}{solrcore}/update?commit=true'
+    headers = {'Content-Type': 'text/xml'}
+    payloadtemplate = '<delete><query>source_id:{sourceid}</query></delete>'
+
+    def requires(self):
+        return Available(service=self.solruri,
+                         message="provide a running Solr, please")
+
+    def run(self):
+        uri = self.uritemplate.format(solruri=self.solruri,
+                                      solrcore=self.solrcore)
+        payload = self.payloadtemplate.format(sourceid=self.sourceid)
+        result = {}
+
+        try:
+            response = requests.post(uri, data=payload, headers=self.headers)
+
+            result['status_code'] = response.status_code
+            result['content'] = response.content.decode('utf-8')
+
+            if result['status_code'] != 200:
+                raise SolrCleanupError('could not successfully clean Solr core \'%s\' (status code = \'%d\')' % (
```

Comment:
For longer comments the [triple-quotes](https://docs.python.org/3/tutorial/introduction.html#strings) are convenient.

> String literals can span multiple lines. One way is using triple-quotes: """...""" or '''...'''

Example:

```python
raise RuntimeError("""could not clean Solr core %s (HTTP %d)""" % (uri, response.status_code))
```